### ISSUE TITLE
Bug: Tests fail to run when using multiple Xcode configurations + test host

### ISSUE THREAD

**Issue by erikkerber** [MobileNativeFoundation/rules_xcodeproj#2178]
### Description

When we enable multiple Xcode configurations and run tests that use our main app as the test host, we observe an Xcode error during launch stating it cannot find the app bundle:

```
Assertions: System: Failed to install or launch the test runner. If you believe this error represents a bug, please attach the result bundle at /Users/sbutterfield/Library/Developer/Xcode/DerivedData/Slack-BwB-edfjnupcrpettshcgylrtgowlszf/Logs/Test/Test-Slack-2023.05.04_16-22-21--0500.xcresult. (Underlying Error: Cannot launch simulated executable: no file found at /Users/sbutterfield/Library/Developer/Xcode/DerivedData/Slack-BwB-edfjnupcrpettshcgylrtgowlszf/Build/Products/Retail-iphonesimulator/bazel-out/applebin_ios-ios_sim_arm64-dbg-ST-b4da387d22bf/bin/App/Slack.app)
```

It appears the transition hash is wrong in `applebin_ios-ios_sim_arm64-dbg-ST-b4da387d22bf`

### Reproduction steps

1. Create an iOS app, rules_xcodeproj setup with multiple configurations
2. Create a test using that app as the test host
3. Run the test

### Expected behavior

It runs

### rules_xcodeproj version

14.1

### Xcode version

14.3

### Bazel version

6.2

### rules_apple version

2.2

### rules_swift version

3fa1b7a364ab9dbb39967763b8859a2d3c6f19aa

**Comment by brentleyjones** [comment]
`examples/integration` has multiple configurations and test hosts, does it repro there? Wondering if there is something more special about your setup (like non-1:1 relationships, or Xcode configurations are “identical”). If it repros with our examples, even “better”.

**Comment by brentleyjones** [comment]
Debugged it offline. I'll have a fix for this soon.

**Comment by erikkerber** [comment]
This appears fixed in 1.6.0. It did not reproduce 100% of the time previously (you could get lucky with the hashes IIRC), so if I was just lucky after 3 regens/tests and this reproduces I'll re-open later.
